[CS2113-W12-4] Java Ninja

naveen42x opened the pull request:
Java Ninja helps beginner programmers improve their understanding of fundamental Java concepts through a CLI-based learning tool.

naveen42x committed:
Fix Checkstyle violations in QuizManager

naveen42x committed:
Normalize line endings to Unix and OS X

naveen42x committed:
Implement help command

naveen42x committed:
Implement help message and fix check errors

naveen42x committed:
Merge pull request #22 from naveen42x/implement-help-command

Implement help command

naveen42x committed:
Implement save and load quiz results functionality

naveen42x committed:
Merge pull request #24 from naveen42x/implement-save-results

Implement save and load quiz results functionality

naveen42x committed:
Updated save feature and fixed test cases

naveen42x committed:
Merge pull request #25 from naveen42x/implement-save-results

Updated save feature and fixed test cases

shaunngoh committed:
Add assertions into Quiz

shaunngoh committed:
Merge pull request #26 from AY2425S1-CS2113-W12-4/Assertions

Add assertions into Quiz

shaunngoh committed:
Resolve QuizManagerTest

shaunngoh committed:
Resolve QuizManagerTest

shaunngoh committed:
Resolve Quiztest

shaunngoh committed:
Merge pull request #27 from AY2425S1-CS2113-W12-4/QuizTest

Resolve QuizManagerTest

Commit:
Add tests, assert, and logging to QuizManager.

Commit:
Fixed checkstyle issues and added logging/assertions.

Commit:
Fixed checkstyle issues.

Commit:
Fixed checkstyle issues.

Commit:
Fixed checkstyle issues.

Commit:
Fixed checkstyle issues.

Commit:
Fixed checkstyle issues.

Lucky-Yuan00 committed:
Merge pull request #29 from Lucky-Yuan00/Jinlin-Dev

Add tests, assert, and logging to QuizManager.

Commit:
Fixed checkstyle issues.

Commit:
Fxi checkstyle issues.

Commit:
Fxi checkstyle issues.

Commit:
Fxi checkstyle issues.

Commit:
Fixed checkstyle issues.

Commit:
Fixed checkstyle issues.

Commit:
Fixed checkstyle issues.

Commit:
Add tests, assert, and logging to QuizManager.

Commit:
Add tests, assert, and logging to QuizManager.

Commit:
Add tests, assert, and logging to QuizManager.

Commit:
Fixed checkstyle issues.

Lucky-Yuan00 committed:
Merge pull request #30 from Lucky-Yuan00/Jinlin-Dev

Fixed checkstyle issues.

shaunngoh committed:
Add Flashcard, user input

shaunngoh committed:
Merge pull request #33 from AY2425S1-CS2113-W12-4/Flashcard

Add Flashcard, user input

naveen42x committed:
Enabled assertions in build.gradle

naveen42x committed:
Merge pull request #34 from naveen42x/enable-assertions

Enabled assertions in build.gradle

naveen42x committed:
Updated QuizManager and test cases

naveen42x committed:
Updated QuizManager and test cases

naveen42x committed:
Merge pull request #35 from naveen42x/enable-assertions

Updated QuizManager and test cases

Commit:
Complete implementation of True/False functionality and input validation

Lucky-Yuan00 committed:
Merge pull request #38 from Lucky-Yuan00/jinlin-dev

Complete implementation of True/False functionality and input validation.

Commit:
Add new test methods and assertions for True/False question handling.

Lucky-Yuan00 committed:
Merge pull request #41 from Lucky-Yuan00/jinlin-dev

Add new test methods and assertions for True/False question handling.

YubotKwng committed:
new question type FITB

enable user to add FITB question in format: add FITB /q[question text] /a[correct answer]

YubotKwng committed:
enable exit

to exit the quiz anytime which only return to the main menu without stopping the progam

YubotKwng committed:
adjust ui-test to pass github tests

YubotKwng committed:
Merge pull request #42 from YubotKwng/Fill-in-the-blank

question type Fill in the blank

YubotKwng committed:
adjust to pass ui-test

YubotKwng committed:
adjustment for ui-test

YubotKwng committed:
class fill in the blank

YubotKwng committed:
ui-test

YubotKwng committed:
Merge pull request #45 from YubotKwng/Fill-in-the-blank

Fill in the blank

YubotKwng committed:
Merge pull request #46 from AY2425S1-CS2113-W12-4/FIll-in-the-blank

Fill in the blank

shaunngoh committed:
Add developer guide architecture

shaunngoh committed:
Merge pull request #48 from AY2425S1-CS2113-W12-4/developer-guide

Add developer guide architecture

shaunngoh committed:
Implement architecture for guide

shaunngoh committed:
Merge pull request #49 from AY2425S1-CS2113-W12-4/developer-guide

Implement architecture for guide

okkhoy commented on src/main/java/seedu/javaninja/Flashcard.java:
```diff
@@ -0,0 +1,27 @@
+    public String getQuestionType () {
+        return "FC";
+    }
+
+
+
+
+
```
unnecessary blank space can be removed.

okkhoy commented on src/main/java/seedu/javaninja/Quiz.java:
```diff
@@ -0,0 +1,84 @@
+            throw new IllegalStateException("Cannot start a quiz with no questions.");
+        }
+
+        while (currentQuestionIndex < questions.size()) {
```
looks quite nested. can you extract some code to individual methods and invoke them to reduce nesting?

okkhoy commented on src/main/java/seedu/javaninja/Quiz.java:
```diff
@@ -0,0 +1,84 @@
+                // Check if the user wants to exit the quiz
+                if (answer.equalsIgnoreCase("exit")) {
+                    System.out.println("Exiting the quiz. Returning to main menu...");
+                    return;  // Exit the quiz early without quitting the entire application
```
redundant comment since the message above is clear

okkhoy commented on src/main/java/seedu/javaninja/QuizManager.java:
```diff
@@ -0,0 +1,270 @@
+        }
+    }
+
+    public void addQuestionByUser(String input) throws IOException {
```
consider extracting code (within each block) to individual methods if possible

okkhoy commented on src/main/java/seedu/javaninja/QuizManager.java:
```diff
@@ -0,0 +1,270 @@
+            String questionLine = "TrueFalse | TrueFalse | " + questionText + " | " + correctAnswer;
+            saveQuestionToFile(questionLine);
+        } else if (input.startsWith("add FITB")) {
+            String[] parts = input.split("/q|/a");
+            if (parts.length < 3) {
+                System.out.println("Invalid command format. Please provide both question and answer.");
+                return;
+            }
```
can be extracted for better slap

okkhoy commented on src/main/java/seedu/javaninja/Storage.java:
```diff
@@ -0,0 +1,71 @@
+package seedu.javaninja;
+
+import java.io.*;
```
don't use wildcard imports

okkhoy reviewed (commented):
Please pay attention to code quality; try to improve it as much as possible before the final submission.
Almost none of the methods have header comments; consider adding header comments to all non-trivial methods.

Commit:
Add TrueFalse class diagram to Developer Guide

Commit:
Complete True/False feature documentation in Developer Guide

Lucky-Yuan00 committed:
Merge pull request #50 from Lucky-Yuan00/jinlin-dev

Add TrueFalse Class Diagram and Documentation in Developer Guide

YubotKwng committed:
delete separate adding question type FITB

YubotKwng committed:
Timer implemented for the quiz

YubotKwng committed:
adjust methods in testing

Commit:
Add saveQuestionToFile method to Storage class to resolve missing method error in QuizManager

Lucky-Yuan00 committed:
Merge pull request #53 from Lucky-Yuan00/jinlin-dev

Add saveQuestionToFile method to Storage class to resolve missing method error in QuizManager

YubotKwng committed:
no message

YubotKwng committed:
adjust error

YubotKwng committed:
no message

YubotKwng committed:
Merge pull request #51 from YubotKwng/Timer

Timer

YubotKwng committed:
Merge branch 'master' into Timer

YubotKwng committed:
no message

YubotKwng committed:
Merge pull request #55 from YubotKwng/Timer

no message

YubotKwng committed:
Merge pull request #54 from AY2425S1-CS2113-W12-4/Timer

Timer

YubotKwng committed:
select number of questions

YubotKwng committed:
add assert

YubotKwng committed:
Merge pull request #56 from YubotKwng/Quiz-number

select number of questions

YubotKwng committed:
Merge pull request #57 from AY2425S1-CS2113-W12-4/QuizNumber

Quiz number

naveen42x committed:
Updated User Guide

naveen42x committed:
Merge pull request #59 from naveen42x/ug1

Updated User Guide

shaunngoh committed:
Add Parser class, refactor Question

shaunngoh committed:
Refactor code, add Parser class

shaunngoh committed:
Merge pull request #60 from AY2425S1-CS2113-W12-4/code-refactor

Add Parser class, refactor Question

shaunngoh committed:
Fix QuizManagerTest class

shaunngoh committed:
Merge pull request #61 from AY2425S1-CS2113-W12-4/code-refactor

Fix QuizManagerTest class

shaunngoh committed:
Include QuizTimer class

shaunngoh committed:
Merge pull request #62 from AY2425S1-CS2113-W12-4/code-refactor

Include QuizTimer class

naveen42x committed:
Updated Developer Guide

naveen42x committed:
Merge pull request #63 from naveen42x/dg

Updated Developer Guide

naveen42x committed:
Updated User Guide

naveen42x committed:
Merge pull request #64 from naveen42x/dg

Updated User Guide

naveen42x committed:
Updated Developer Guide

naveen42x committed:
Merge pull request #65 from naveen42x/dg

Updated Developer Guide

YubotKwng committed:
UserGuide update

YubotKwng committed:
Adjust error

YubotKwng committed:
adjust for issue

naveen42x committed:
Updated Developer Guide

YubotKwng committed:
adjustment on failure test

naveen42x committed:
Merge pull request #67 from naveen42x/dg

Updated Developer Guide

YubotKwng committed:
Merge pull request #66 from YubotKwng/master

UserGuide update

naveen42x committed:
Added Topic class diagram

naveen42x committed:
Added Topic class diagram

naveen42x committed:
Merge pull request #68 from naveen42x/dg

Added Topic class diagram link in DG

Alfred-Goh02 commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,145 @@
+
+The Sequence Diagram below shows how the components interact with each other for the scenario where the user issues the command `add Flashcard /q what's 2+3 /a 5`
+
+![image](https://github.com/user-attachments/assets/1dd443de-b538-4fb8-b4d1-3bff4643b3dd)
```
Please include the picture in your repository and link it through your repository instead.

Alfred-Goh02 replied:
Consider using less colours for your sequence diagrams, and use normal rectangle instead of rounded ones

Alfred-Goh02 commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,145 @@
+
+#### UML Class Diagram
+
+![TrueFalse Class Diagram](https://github.com/Lucky-Yuan00/tp/blob/jinlin-dev/docs/UML/TrueFalseClassDiagram.png?raw=true)
```
For your true false diagram, use + - on the attributes to indicate access modifiers

yijiano replied:
Maybe consider using the standards that were mentioned in the textbook instead of using arbitrary symbols.

Alfred-Goh02 commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,145 @@
+
+#### UML Class Diagram
+
+![TrueFalse Class Diagram](https://github.com/Lucky-Yuan00/tp/blob/jinlin-dev/docs/UML/TrueFalseClassDiagram.png?raw=true)
```
Do not have to put C circle, it is implied it is a class

andreusxcarvalho commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,145 @@
+This tool will offer an interactive learning experience through a series of tasks and challenges designed to deepen users' understanding and proficiency in both command-line operations and Java language.
 
 ## User Stories
 
```
Are there more user stories?

andreusxcarvalho commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,145 @@
+How the architecture components interact with each other: 
+
+The Sequence Diagram below shows how the components interact with each other for the scenario where the user issues the command `add Flashcard /q what's 2+3 /a 5`
+
```
Since you gave a sequence diagram for add Flashguard, you should have a flow of events (for that feature) above it to explain what goes on.

andreusxcarvalho commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,145 @@
+![image](https://github.com/user-attachments/assets/1dd443de-b538-4fb8-b4d1-3bff4643b3dd)
+
+
+### True/False Feature Implementation
```
Maybe you could rename it CheckAnswer Feature

andreusxcarvalho commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,145 @@
+     Displays answer choices ("1. True" and "2. False") for the user. This makes it clear what options are available for true/false questions.
+   - `toString()`:
+     Returns a string representation of the question with a "(True/False)" label to indicate its type.
+
```
Good work on giving the flow of events for the CheckAnswer feature, do add on the UML sequence diagram for that feature

shaunngoh committed:
Add developer guide

shaunngoh committed:
Merge pull request #70 from AY2425S1-CS2113-W12-4/dev-guide

Add developer guide

YubotKwng committed:
no message

YubotKwng committed:
add DG

YubotKwng committed:
Add DG

YubotKwng committed:
Merge pull request #71 from YubotKwng/master

Add DG for FITB

KuanHsienn commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,264 @@
+
+#### UML Class Diagram
+
+![FillInTheBlank UML Class Diagram](https://github.com/YubotKwng/tp/blob/e3358e1cc869012c202864071fbb75f2f8261712/docs/UML/FITBClassDiagram.png)
```
You might want to check if the PNG link is valid if the picture is already ready

seanngja replied:
The diagram is not rendering in the web page.

yijiano replied:
This is a broken link. Please consider adding the image assets to the repo itself to minimize such issues.

ctzeyong replied:
Might want to check if the diagram renders properly on the web page?

seanngja commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,264 @@
+      - Pros: Increases code modularity and allows for more scoring logic.
+      - Cons: Adds complexity that may not be necessary if the scoring logic remain simple.
+
+#### UML Class Diagram
+Below is the class diagram for the `QuizManager` class, illustrating its attributes and methods.
+![image](https://github.com/user-attachments/assets/85b30191-76ab-431f-8cdb-0e29c481e506)
+
```
Perhaps including a sequence diagram here would be more meaningful? The class diagram alone does not really give much insight into how QuizManager works.

seanngja commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,264 @@
+
+#### UML Class Diagram
+Below is the class diagram for the `QuizManager` class, illustrating its attributes and methods.
+![image](https://github.com/user-attachments/assets/85b30191-76ab-431f-8cdb-0e29c481e506)
```
Maybe you could add the associations between the classes that it interacts with.

yijiano replied:
Moreover, was there a need for the class diagram when you have listed out the class details in front in point form. Alternatively, I agree with @seanngja 's suggestion to show class associations.

KuanHsienn commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,264 @@
+      Adds a `Question` object to the topic’s question list.
+    - `List<Question> getQuestions()`:  
+      Returns the list of questions under the topic.
+    - `String getName()`:  
```
You might want to consider standardising the type of bullet points (line 236) used to retain consistency.

seanngja commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,264 @@
+#### **Feature Overview**
+The `Topic` class encapsulates a programming topic along with its related questions. It helps in grouping relevant questions, enabling quizzes to be topic-based, and provides essential methods to add and access questions within each topic.
+
+#### **Implementation Details**
+
+1. **Attributes**:
+    - `String name`: Stores the name of the topic.
+    - `List<Question> questions`: Holds a list of questions related to this topic.
+
+2. **Constructor**:
+    - `Topic(String name)`:  
+      Initializes the topic with a specified name and creates an empty list to hold its questions. Ensures that the `name` is non-null to prevent invalid topics.
+
+3. **Methods**:
+    - `addQuestion(Question question)`:  
+      Adds a `Question` object to the topic’s question list.
+    - `List<Question> getQuestions()`:  
+      Returns the list of questions under the topic.
+    - `String getName()`:  
+      Returns the name of the topic as a string.
```
Is this really necessary? I feel that the implementation details becomes rather lengthy and repetitive as these are already being showcased in your class diagrams.

KuanHsienn commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,264 @@
+
+#### UML Class Diagram
+Below is the class diagram for the `QuizManager` class, illustrating its attributes and methods.
+![image](https://github.com/user-attachments/assets/85b30191-76ab-431f-8cdb-0e29c481e506)
```
For images you might want to consider using relative path to the image directory instead of absolute path

seanngja reviewed (commented):
I think more sequence diagrams in general would help showcase your implementation more.

KuanHsienn commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,264 @@
+
+1. **Attributes**:
+   - Constants:
+     - `QUESTIONS_FILE_PATH`: Specifies the path to the file that stores quiz questions.
+     - `RESULTS_FILE_PATH`: Specifies the path to the file that stores past quiz results.
+   - Core Components:
+     - `List<Topic> topics`: Stores all available quiz topics.
+     - `Quiz currentQuiz`: Represents the quiz currently in progress.
+     - `List<String> pastResults`: Maintains a list of past quiz results, with each result stored as a formatted string.
+   - Data Storage:
+     - `Storage results`: Manages loading and saving of past quiz results.
+     - `Storage questions`: Manages loading and saving of quiz questions.
```
Maybe consider listing the attributes constant throughout? As some has variable type but some do not

yijiano commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,264 @@
 
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
```
Should this placeholder be removed since you have already added the necessary acknowledgements below?

yijiano commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,264 @@
+- Code adapted from [AddressBook-Level3](https://github.com/se-edu/addressbook-level3).
+
+JavaNinja uses the following tools for development:
+- JUnit - Used for testing.
```
Was there a need to repeat the mention of JUnit test here?

yijiano commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,264 @@
+
+`Commons` represent a collection of classes used by other components.
+
+How the architecture components interact with each other: 
```
Should this be made into a header to increase clarity?

yijiano commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,264 @@
+Main components of the architecture:
+
+`Main` (consisting of class `JavaNinja`) is in charge of the app launch and shut down. 
+1. At app launch, it initialises the components in the correct order and connects them with each other
```
Perhaps there could have been more illustration to show the correct order of initialisation.

yijiano commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,264 @@
+
+#### UML Class Diagram
+Below is the class diagram for the `Topic` class, illustrating its attributes and methods.
+![Topic Class Diagram](https://github.com/naveen42x/tp/blob/dg/docs/UML/TopicClasDiagram.png?raw=true)
```
Please refer to the comments given by all the other reviewers for QuizManager.

ctzeyong commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,264 @@
+
+#### UML Class Diagram
+
+![TrueFalse Class Diagram](https://github.com/Lucky-Yuan00/tp/blob/jinlin-dev/docs/UML/TrueFalseClassDiagram.png?raw=true)
```
Can check if the notation is in line with what we use in this course?
C circle is not needed and +/- signs should be used for public/private attributes or methods.

ctzeyong replied:
Should the extends label be removed from the arrow? The arrow from TrueFalse to Question class should already mean TrueFalse extends Question.

yijiano commented on docs/DeveloperGuide.md:
```diff
@@ -36,3 +281,9 @@
+### Manual Testing
+View the [User Guide](https://AY2425S1-CS2113-W12-4.github.io/tp/UserGuide.html) for the full list of UI commands and their related use case and expected outcomes.
+
+### JUnit Testing
```
Perhaps you can give some instructions, or even include the gradle tests.

yijiano commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,264 @@
 ### Target user profile
 
-{Describe the target user profile}
+Beginner programmers looking to improve their proficiency in Java and command-line skills.
```
Perhaps you can expand on this together with your user stories.

yijiano reviewed (commented):
Looks good overall given that it is mostly incomplete, some nits to take note of.

ctzeyong commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,264 @@
-{Describe the value proposition: what problem does it solve?}
+This tool will offer an interactive learning experience through a series of tasks and challenges designed to deepen users' understanding and proficiency in both command-line operations and Java language.
 
 ## User Stories
```
Might want to check if user stories are updated?

ctzeyong commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,264 @@
+    - Allow users to review their scores and comments from previous quiz attempts, providing a way to track their progress.
+
+
+#### **Implementation Details**
```
Perhaps you could include a class diagram that shows the associations between QuizManager class and other classes?

ctzeyong commented on docs/DeveloperGuide.md:
```diff
@@ -1,22 +1,264 @@
+![image](https://github.com/user-attachments/assets/85b30191-76ab-431f-8cdb-0e29c481e506)
+
+
+### True/False Feature Implementation
```
Perhaps can consider adding a section to explain the Question class that the TrueFalse, FillInTheBlank classes extend from?

ctzeyong reviewed (commented):
First portion of the DG is generally well explained. Good job!
However, the back portion including user stories, exceptions and logging, manual testing does not seem to be updated.

Commit:
Updated UG

Lucky-Yuan00 committed:
Merge pull request #72 from Lucky-Yuan00/jinlin-dev

Updated UG

Commit:
Updated DG

Lucky-Yuan00 committed:
Merge pull request #73 from Lucky-Yuan00/jinlin-dev

Updated DG

shaunngoh committed:
Add QuizResults, QuizSession class

shaunngoh committed:
Merge pull request #74 from AY2425S1-CS2113-W12-4/more-oop

Add QuizResults, QuizSession class

shaunngoh committed:
Adjust error in QuizManagerTest

shaunngoh committed:
Merge pull request #75 from AY2425S1-CS2113-W12-4/more-oop

Adjust error in QuizManagerTest

shaunngoh committed:
Add developer details for QuizManager, Parser and JavaNinja

shaunngoh committed:
Merge pull request #76 from AY2425S1-CS2113-W12-4/add-dev-guide

Add developer details for QuizManager, Parser and JavaNinja

shaunngoh committed:
Add additional developer details

shaunngoh committed:
Merge pull request #77 from AY2425S1-CS2113-W12-4/add-dev-guide

Add additional developer details

shaunngoh committed:
Add User Guide details

shaunngoh committed:
Merge pull request #78 from AY2425S1-CS2113-W12-4/add-user-guide

Add User Guide details

shaunngoh committed:
Add additional test classes

shaunngoh committed:
Add methods to test classes
	modified:   ../src/main/java/seedu/javaninja/Quiz.java
	modified:   ../src/main/java/seedu/javaninja/QuizSession.java
	modified:   ../src/test/java/seedu/javaninja/QuizResultsTest.java
	modified:   ../src/test/java/seedu/javaninja/QuizSessionTest.java
	modified:   ../src/test/java/seedu/javaninja/StorageTest.java

shaunngoh committed:
Merge pull request #80 from AY2425S1-CS2113-W12-4/quiz-test

Add additional test classes

shaunngoh committed:
Improve Code Quality

	modified:   src/main/java/seedu/javaninja/QuizManager.java
	modified:   src/main/java/seedu/javaninja/QuizResults.java
	modified:   src/main/java/seedu/javaninja/QuizSession.java
	modified:   src/main/java/seedu/javaninja/QuizTimer.java
	modified:   src/main/java/seedu/javaninja/Storage.java
	modified:   src/main/java/seedu/javaninja/question/Flashcard.java
	modified:   src/main/java/seedu/javaninja/question/Mcq.java
	modified:   src/main/java/seedu/javaninja/question/Question.java
	modified:   src/test/java/seedu/javaninja/QuizManagerTest.java
	modified:   src/test/java/seedu/javaninja/QuizResultsTest.java

shaunngoh committed:
Improve code quality

	modified:   ../src/main/java/seedu/javaninja/QuizTimer.java
	modified:   ../src/main/java/seedu/javaninja/Topic.java
	modified:   ../src/main/java/seedu/javaninja/TopicManager.java

shaunngoh committed:
Merge pull request #81 from AY2425S1-CS2113-W12-4/quiz-test

Improve Code Quality

shaunngoh committed:
Add TopicManager and Topic Initialisation in dev
	modified:   ../docs/DeveloperGuide.md
	modified:   ../docs/UML/ParserSequenceDiagram.png
	modified:   ../docs/UML/QuizManagerSequenceDiagram.png
	renamed:    ../docs/UML/puml-files/QuizManagerArchitecture.puml -> ../docs/UML/puml-files/QuizManagerClassHighLevel.puml
	modified:   ../docs/UML/puml-files/QuizManagerSequenceDiagram.puml
	new file:   ../docs/UML/puml-files/TopicInstantiationSequenceDiagram.puml
	new file:   ../docs/UML/puml-files/TopicManager.puml
	modified:   ../src/main/java/seedu/javaninja/Quiz.java
	modified:   ../src/main/java/seedu/javaninja/TopicManager.java

shaunngoh committed:
Merge pull request #82 from AY2425S1-CS2113-W12-4/developer-guide-details

Add TopicManager and Topic Initialisation in dev

shaunngoh committed:
Add cli dependability to classes

	modified:   ../data/Questions.txt
	modified:   ../src/main/java/seedu/javaninja/Cli.java
	modified:   ../src/main/java/seedu/javaninja/JavaNinja.java
	modified:   ../src/main/java/seedu/javaninja/Parser.java
	modified:   ../src/main/java/seedu/javaninja/Quiz.java
	modified:   ../src/main/java/seedu/javaninja/QuizManager.java
	modified:   ../src/main/java/seedu/javaninja/QuizSession.java
	modified:   ../src/main/java/seedu/javaninja/TopicManager.java
	modified:   ../src/main/java/seedu/javaninja/question/FillInTheBlank.java
	modified:   ../src/main/java/seedu/javaninja/question/Flashcard.java
	modified:   ../src/main/java/seedu/javaninja/question/Mcq.java
	modified:   ../src/main/java/seedu/javaninja/question/Question.java
	modified:   ../src/main/java/seedu/javaninja/question/TrueFalse.java
	modified:   ../src/test/java/seedu/javaninja/QuizManagerTest.java
	modified:   ../src/test/java/seedu/javaninja/QuizSessionTest.java

	../docs/UML/TopicInstantiationSequenceDiagram.png
	../docs/UML/TopicManager.png

shaunngoh committed:
Ensure gradlew testability
	modified:   data/Questions.txt
	modified:   src/main/java/seedu/javaninja/Quiz.java
	modified:   src/main/java/seedu/javaninja/QuizSession.java
	modified:   src/test/java/seedu/javaninja/QuizManagerTest.java
	modified:   src/test/java/seedu/javaninja/QuizSessionTest.java

shaunngoh committed:
Merge pull request #83 from AY2425S1-CS2113-W12-4/cli--development

Add cli dependability to classes

shaunngoh committed:
Add RandomQuestion Method, improve QuizResults
	modified:   ../data/Questions.txt
	modified:   ../src/main/java/seedu/javaninja/Quiz.java
	modified:   ../src/main/java/seedu/javaninja/QuizManager.java
	modified:   ../src/main/java/seedu/javaninja/QuizResults.java
	modified:   ../src/main/java/seedu/javaninja/QuizSession.java
	modified:   ../src/main/java/seedu/javaninja/Topic.java
	modified:   ../src/main/java/seedu/javaninja/TopicManager.java
	modified:   ../src/main/java/seedu/javaninja/question/Mcq.java
	modified:   ../src/main/java/seedu/javaninja/question/Question.java
	modified:   ../src/test/java/seedu/javaninja/QuizResultsTest.java
	modified:   ../src/test/java/seedu/javaninja/QuizSessionTest.java
	modified:   EXPECTED.TXT

	../docs/UML/TopicInstantiationSequenceDiagram.png
	../docs/UML/TopicManager.png

shaunngoh committed:
Comply with gradlew check

shaunngoh committed:
Merge pull request #84 from AY2425S1-CS2113-W12-4/cli--development

Add RandomQuestion Method, improve QuizResults

shaunngoh committed:
Fix bugs in time limit and Flashcard

shaunngoh committed:
Comply with gradlew

shaunngoh committed:
Comply with gradlew check

shaunngoh committed:
Merge pull request #85 from AY2425S1-CS2113-W12-4/bug--fixing

Fix bugs in time limit and Flashcard

naveen42x committed:
Create PPP

naveen42x committed:
Merge pull request #88 from naveen42x/PPP1

Create PPP

shaunngoh committed:
Furnish details for developer guide
	modified:   ../../../docs/DeveloperGuide.md
	deleted:    ../../../docs/UML/ParserSequenceDiagram.png
	new file:   ../../../docs/UML/puml-files/Cli/CliClassDiagram.puml
	renamed:    ../../../docs/UML/FITBClassDiagram.png -> ../../../docs/UML/puml-files/Fitb/FITBClassDiagram.png
	renamed:    ../../../docs/UML/puml-files/FillInTheBlank.puml -> ../../../docs/UML/puml-files/Fitb/FillInTheBlank.puml
	renamed:    ../../../docs/UML/JavaNinja.png -> ../../../docs/UML/puml-files/JavaNinja/JavaNinja.png
	renamed:    ../../../docs/UML/puml-files/JavaNinja.puml -> ../../../docs/UML/puml-files/JavaNinja/JavaNinja.puml
	new file:   ../../../docs/UML/puml-files/JavaNinja/JavaNinjaSequenceDiagram.puml
	new file:   ../../../docs/UML/puml-files/Parser/ParserSequenceDiagram.png
	renamed:    ../../../docs/UML/puml-files/ParserSequenceDiagram.puml -> ../../../docs/UML/puml-files/Parser/ParserSequenceDiagram.puml
	renamed:    ../../../docs/UML/QuizManager.png -> ../../../docs/UML/puml-files/QuizManager/QuizManager.png
	renamed:    ../../../docs/UML/QuizManagerAndQuizRelations.png -> ../../../docs/UML/puml-files/QuizManager/QuizManagerAndQuizRelations.png
	renamed:    ../../../docs/UML/puml-files/QuizManagerAndQuizRelations.puml -> ../../../docs/UML/puml-files/QuizManager/QuizManagerAndQuizRelations.puml
	renamed:    ../../../docs/UML/QuizManagerArchitecture.png -> ../../../docs/UML/puml-files/QuizManager/QuizManagerArchitecture.png
	renamed:    ../../../docs/UML/puml-files/QuizManagerClassDiagram.puml -> ../../../docs/UML/puml-files/QuizManager/QuizManagerClassDiagram.puml
	renamed:    ../../../docs/UML/puml-files/QuizManagerClassHighLevel.puml -> ../../../docs/UML/puml-files/QuizManager/QuizManagerClassHighLevel.puml
	renamed:    ../../../docs/UML/QuizManagerSequenceDiagram.png -> ../../../docs/UML/puml-files/QuizManager/QuizManagerSequenceDiagram.png
	renamed:    ../../../docs/UML/puml-files/QuizManagerSequenceDiagram.puml -> ../../../docs/UML/puml-files/QuizManager/QuizManagerSequenceDiagram.puml
	new file:   ../../../docs/UML/puml-files/QuizSession/QuizSessionClassDiagram.puml
	new file:   ../../../docs/UML/puml-files/QuizSession/QuizSessionSequenceDiagram.puml
	renamed:    ../../../docs/UML/TopicClasDiagram.png -> ../../../docs/UML/puml-files/TopicManager/TopicClasDiagram.png
	renamed:    ../../../docs/UML/TopicInstantiationSequenceDiagram.png -> ../../../docs/UML/puml-files/TopicManager/TopicInstantiationSequenceDiagram.png
	renamed:    ../../../docs/UML/puml-files/TopicInstantiationSequenceDiagram.puml -> ../../../docs/UML/puml-files/TopicManager/TopicInstantiationSequenceDiagram.puml
	renamed:    ../../../docs/UML/TopicManager.png -> ../../../docs/UML/puml-files/TopicManager/TopicManager.png
	renamed:    ../../../docs/UML/puml-files/TopicManager.puml -> ../../../docs/UML/puml-files/TopicManager/TopicManager.puml
	renamed:    ../../../docs/UML/TrueFalseClassDiagram.png -> ../../../docs/UML/puml-files/TrueFalse/TrueFalseClassDiagram.png
	renamed:    ../../../docs/UML/puml-files/TrueFalseClassDiagram.puml -> ../../../docs/UML/puml-files/TrueFalse/TrueFalseClassDiagram.puml
	../../../docs/UML/puml-files/Cli/CliClassDiagram.png
	../../../docs/UML/puml-files/JavaNinja/JavaNinjaSequenceDiagram.png
	../../../docs/UML/puml-files/QuizSession/QuizSessionClassDiagram.png
	../../../docs/UML/puml-files/QuizSession/QuizSessionSequenceDiagram.png

shaunngoh committed:
Merge pull request #90 from AY2425S1-CS2113-W12-4/add-details-dg

Furnish details for developer guide

shaunngoh committed:
Add missing photo

shaunngoh committed:
Merge pull request #91 from AY2425S1-CS2113-W12-4/add-details-dg

Add missing photo

shaunngoh committed:
Add user guide details

shaunngoh committed:
Merge pull request #92 from AY2425S1-CS2113-W12-4/add-details-dg

Add user guide details

YubotKwng committed:
Cli Sequence diagram

YubotKwng committed:
Merge pull request #160 from YubotKwng/master

Cli Sequence diagram

YubotKwng committed:
Test for Cli class

YubotKwng committed:
Checkstyle fixing

YubotKwng committed:
Merge pull request #161 from YubotKwng/master

Test for Cli class

YubotKwng committed:
Test for TopicManager

YubotKwng committed:
checkstyle fix

YubotKwng committed:
Merge pull request #162 from YubotKwng/master

Test for TopicManager

YubotKwng committed:
test for FITB

YubotKwng committed:
Merge pull request #163 from YubotKwng/master

test for FITB

YubotKwng committed:
Test for Parser

YubotKwng committed:
Merge pull request #164 from YubotKwng/master

Test for Parser

YubotKwng committed:
Header Comments for methods for classes: Storage, Topic, TopicManager

YubotKwng committed:
Merge pull request #165 from YubotKwng/master

Header Comments for methods for classes: Storage, Topic, TopicManager

Commit:
Implement class and sequence diagrams for QuizSession section

Lucky-Yuan00 committed:
Merge pull request #167 from Lucky-Yuan00/jinlin-dev

Implement and improve class and sequence diagrams for QuizSession section.

Commit:
Improve JUnit Testing all the tests and complete the description in the DG.

Lucky-Yuan00 committed:
Merge pull request #168 from Lucky-Yuan00/jinlin-dev

Improve JUnit Testing all the tests and complete the description in the DG.

Commit:
Added Display time left feature and added comments

Commit:
Modify formatting issues

Commit:
Modify formatting issues

Lucky-Yuan00 committed:
Merge pull request #170 from Lucky-Yuan00/jinlin-dev

Added display time left feature and added many comments.

Commit:
Implement Project Portfolio Page (PPP)

Lucky-Yuan00 committed:
Merge pull request #171 from Lucky-Yuan00/jinlin-dev

Implement Project Portfolio Page (PPP).

naveen42x committed:
update comments and topics

naveen42x committed:
Merge pull request #172 from naveen42x/update-topic-and-comments

Update comments and topics

naveen42x committed:
Create QuizTimerTest

naveen42x committed:
Merge pull request #173 from naveen42x/update-topic-and-comments

Create QuizTimerTest

naveen42x committed:
Fix checkstyle error

naveen42x committed:
Merge pull request #174 from naveen42x/update-topic-and-comments

Fix checkstyle error

naveen42x committed:
Add more topics

naveen42x committed:
Merge pull request #175 from naveen42x/update-topic-and-comments

Add more topics

naveen42x committed:
fix error to pass checks

naveen42x committed:
Merge pull request #176 from naveen42x/update-topic-and-comments

Fix error to pass checks

shaunngoh committed:
Add quizGenerator class, incomplete implementation
	new file:   UML/puml-files/Cli/CliClassDiagram.png
	new file:   UML/puml-files/JavaNinja/JavaNinjaSequenceDiagram.png
	modified:   ../src/main/java/seedu/javaninja/Parser.java
	modified:   ../src/main/java/seedu/javaninja/Quiz.java
	new file:   ../src/main/java/seedu/javaninja/QuizGenerator.java
	modified:   ../src/main/java/seedu/javaninja/QuizManager.java
	deleted:    ../src/main/java/seedu/javaninja/QuizSession.java
	modified:   ../src/main/java/seedu/javaninja/TopicManager.java

naveen42x committed:
Merge pull request #177 from AY2425S1-CS2113-W12-4/add-quizGenerator

Add quizGenerator class, incomplete implementation

YubotKwng committed:
Yubo's PPP

YubotKwng committed:
Merge pull request #178 from YubotKwng/master

Yubo's PPP

naveen42x committed:
Add more comments

naveen42x committed:
Merge pull request #179 from naveen42x/update-topic-and-comments

Add more comments

naveen42x committed:
Update DG

naveen42x committed:
Merge pull request #180 from naveen42x/update-topic-and-comments

Update DG

shaunngoh committed:
Modify quizGenerator class

shaunngoh committed:
Merge pull request #181 from AY2425S1-CS2113-W12-4/quizGenerator

Modify quizGenerator class

shaunngoh committed:
Fix bugs from PE dry run

shaunngoh committed:
Merge pull request #182 from AY2425S1-CS2113-W12-4/fix--bugs

Fix bugs from PE dry run

YubotKwng committed:
Issue solved #107

YubotKwng committed:
fix issue Hyperlinks in UG broken #95

YubotKwng committed:
fix git test

YubotKwng committed:
no message

YubotKwng committed:
modify expected.txt to pass git test

YubotKwng committed:
ui-test

YubotKwng committed:
ui-test modify

YubotKwng committed:
Ui-test modify

YubotKwng committed:
Merge pull request #183 from YubotKwng/master

Issue solved #107

naveen42x committed:
Improve and expand review results command

naveen42x committed:
Merge pull request #184 from naveen42x/v2.1n

Improve and expand review results command

naveen42x committed:
Fix Bugs

naveen42x committed:
Merge pull request #185 from naveen42x/v2.1n

Fix Bugs

shaunngoh committed:
Fix select random bug

naveen42x committed:
Fix more bugs

naveen42x committed:
Merge pull request #186 from naveen42x/v2.1n

Fix more bugs

naveen42x committed:
Fix bugs

naveen42x committed:
Merge pull request #187 from naveen42x/v2.1n

Fix bugs

naveen42x committed:
Fix bug with random quiz feature

naveen42x committed:
Merge pull request #188 from naveen42x/v2.1n

Fix bug with random quiz feature

shaunngoh committed:
Add shaun's project portfolio

shaunngoh committed:
Merge pull request #189 from AY2425S1-CS2113-W12-4/project-portfolio

Add shaun's project portfolio

Commit:
Fix Topic Manager DG sequence diagram.

Lucky-Yuan00 committed:
Merge pull request #190 from Lucky-Yuan00/jinlin-dev

Fix Topic Manager DG sequence diagram.

naveen42x committed:
pass checks

naveen42x committed:
Merge pull request #191 from naveen42x/v2.1n

pass checks

shaunngoh committed:
Add additional PPP details

shaunngoh committed:
Merge pull request #192 from AY2425S1-CS2113-W12-4/project-portfolio

Add additional PPP details

Commit:
Finish making the QuizResults class and the sequence diagram.

Lucky-Yuan00 committed:
Merge pull request #193 from Lucky-Yuan00/jinlin-dev

Finish making the QuizResults class and the sequence diagram.

naveen42x committed:
pass checks

naveen42x committed:
Merge pull request #194 from naveen42x/v2.1n

pass checks

Commit:
Add PPP detail

Lucky-Yuan00 committed:
Merge pull request #195 from Lucky-Yuan00/jinlin-dev

Add PPP detail.

naveen42x committed:
Update Docs

naveen42x committed:
Merge branch 'master' into v2.1n

naveen42x committed:
Merge pull request #197 from naveen42x/v2.1n

Update Docs

shaunngoh committed:
Delete results
	deleted:    data/results.txt

shaunngoh committed:
Fix merge

naveen42x committed:
Update UG

naveen42x committed:
v2.1

naveen42x committed:
Merge pull request #198 from naveen42x/v2.1n

V2.1n

naveen42x committed:
Update UG

naveen42x committed:
Merge pull request #199 from naveen42x/v2.1n

Update UG

Commit:
Add previously removed code

shaunngoh committed:
Merge remote-tracking branch 'refs/remotes/origin/master'

Lucky-Yuan00 committed:
Merge pull request #200 from Lucky-Yuan00/jinlin-dev

Add previously removed code.

shaunngoh committed:
Fix gradlew checks
	modified:   src/main/java/seedu/javaninja/QuizGenerator.java
	modified:   src/main/java/seedu/javaninja/QuizResults.java

shaunngoh committed:
Fix runtest.sh

shaunngoh committed:
fix gradlew checks

shaunngoh committed:
Merge pull request #201 from AY2425S1-CS2113-W12-4/fix--gradlew

Fix  gradlew

YubotKwng committed:
fix for hyperlink

YubotKwng committed:
Merge pull request #202 from YubotKwng/master

fix for hyperlink
